src/support/intervals.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Helpers for handling a generic range of values
+
+#ifndef wasm_support_intervals_h
+#define wasm_support_intervals_h
+
+#include <set>
+#include <vector>
+
+namespace wasm {
+
+struct Interval {
+  unsigned start;
+  unsigned end;
+  // The weight is used to determine which interval to keep when two overlap,
+  // higher is better
+  unsigned weight;
+  Interval(unsigned start, unsigned end, unsigned weight)
+    : start(start), end(end), weight(weight) {}
+
+  bool operator<(const Interval& other) const {
+    return start < other.start && weight < other.weight;
+  }
```

Comment:
This should take `end` into account as well. Otherwise the `std::set<Interval>` returned by `IntervalProcessor::getOverlaps()` will not be able to hold two intervals that differ only in their ends.

Reply:
done